php/class-customize-snapshot.php

```diff
@@ -296,9 +296,45 @@ public function values() {
 
 		// Filter when the scope is dirty.
 		if ( $dirty ) {
-			$values = array_filter( $values, function( $setting ) use ( $dirty ) {
+			$dirty_values = array_filter( $values, function( $setting ) use ( $dirty ) {
 				return $setting['dirty'] === $dirty;
 			} );
+
+			$customize_widgets = $this->snapshot_manager->customize_manager->widgets;
+			$dirty_widgets = array();
+
+			// Find widgets in the dirty sidebars.
+			foreach ( $dirty_values as $setting_id => $setting ) {
+				if ( -1 !== strpos( $setting_id, 'sidebars_widgets' ) ) {
+					$dirty_widgets += (array) $setting['value'];
```

Comment:
Using `+` for array concatenation yields unexpected results. I think you actually mean `array_merge()` here? See for example: https://3v4l.org/50WYF